### PR TITLE
[DEVOPS-688]: AWS Event Message formatting & removal of VPC settings

### PR DESCRIPTION
**Pull request by dannyibishev**
## Changes in this PR.

- Removes the VPC attributes from the serverless file. There is no reason our lambda and SNS topics need to be scoped to a VPC.
- Added support for tidying up AWS Events. Many thanks to @jimmy718 for the help :)

- Tested locally.
- Integration Tests passing in our channel post-deployment to dev.

Screenshot 2022-03-04 at 10 31 45

### CODE REVIEW

File: src/interfaces/slack.d.ts [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 
 export interface SlackMessageAttachment {
+    title: string;
+    fallback: string;
+    text: string;
+    color: string;
+    ts: string;
+}
+
+export interface SlackMessageAttachmentWithLink {
```
Comment (oaklees) [inline src/interfaces/slack.d.ts:14]: We could simplify this to reduce duplication:

```ts
export interface SlackMessageAttachmentWithLink extends SlackMessageAttachment {
    title_link: string;
}
```

File: src/lib/slack.ts [hunk 2]
```diff
@@ -156,6 +156,67 @@ export function formatEventDMSMessage(message: EventDMSMessage, subject: string)
     };
 }
 
+/**
+ * Format an AWS Health event message for Slack.
```
Comment (oaklees) [inline src/lib/slack.ts:160]: This file's going to need a refactor soon - probably benefit from some kind of factory approach to resolve formatters as first class objects vs functions within this same file, we could then extend functionality without having to keep changing the same file.

File: src/sns.ts [hunk 3]
```diff
@@ -23,6 +23,12 @@ export function handler(event: SNSEvent, context: Context, callback: Callback):
         slackMessage = formatEventDMSMessage(message, subject);
     } else if (message.hasOwnProperty('Event Source')) {
         slackMessage = formatEventMessage(message);
+    } else if (message.hasOwnProperty('eventArn')) {
+        if (message.eventArn.includes('arn:aws:health')) {
+            slackMessage = formatEventAWSHealthMessage(message, snsTimestamp);
+        } else {
+            slackMessage = defaultEventMessage(JSON.stringify(message), subject, snsTimestamp);
+        }
```
Comment (oaklees) [inline src/sns.ts:31]: This is all getting a bit gnarly and could also do with a refactor at some point.

File: src/lib/slack.ts [hunk 4]
```diff
@@ -156,6 +156,67 @@ export function formatEventDMSMessage(message: EventDMSMessage, subject: string)
...
+ * @param {EventAWSHealthMessage} message
+ * @param string sns_timestamp
+ * @returns {SlackMessage}
+ */
+export function formatEventAWSHealthMessage(message: EventAWSHealthMessage, snsTimestamp: string): SlackMessage {
+    const timestamp: string = (new Date(snsTimestamp).getTime() / 1000).toFixed(0);
+    const summary: string = 'This is an AWS Event notification';
```
Comment (oaklees) [inline src/lib/slack.ts:167]: Probably don't need the string types here as it's inferred from the initialisation.

**Review by oaklees — APPROVED** [review]

**Comment by jimmy718** [comment]
@dannyibishev Happy to help out on any refactoring when the time comes 🤙

**Pull request merged**